File: Python/Problem Solving/Codewars Problem Solving/8 kyu/flick_switch.py
#  Flick Switch
# *  My Solution


def flick_switch(arr):
    booleans = []
    should_return_true = True

    for element in arr:
        if element == "flick":
            should_return_true = not should_return_true

        booleans.append(should_return_true)

    return booleans


print(flick_switch(["codewars", "flick", "code", "wars"]))

File: c/problem-solving/codewars-problem-solving/8-kyu/flick-switch.c
// Flick Switch
//* My Solution

#include <stdbool.h>
#include <stddef.h>

void flick_switch(size_t length, const char *const array[length], bool result[length])
{
  bool should_return_true = true;

  for (size_t i = 0; i < length; i++)
  {
    if (array[i] && strcmp(array[i], "flick") == 0)
    {
      should_return_true = !should_return_true;
    }

    result[i] = should_return_true;
  }
}
